main.cpp

```diff
@@ -21,79 +21,98 @@
 #endif
 
 #define SAMPLE_TIME     10 // milli-sec
+#define COMPLETED_FLAG (1UL << 0)
+
+PlatformMutex stdio_mutex;
+EventFlags threadFlag;
 
 void print_stats()
 {
     mbed_stats_socket_t stats[MBED_CONF_NSAPI_SOCKET_STATS_MAX_COUNT];
-    static int num = 0;
-    int count;
+    static size_t num = 0;
+    size_t count;
     
     memset(&stats[0], 0, sizeof(mbed_stats_socket_t) * MBED_CONF_NSAPI_SOCKET_STATS_MAX_COUNT);
     printf("%-15s%-15s%-15s%-15s%-15s%-15s%-15s\n", "Num", "ID", "State", "Proto", "Sent", "Recv", "Time");
-    while (1) {
-
+    while (COMPLETED_FLAG != threadFlag.get()) 
```

Comment:
```suggestion
    while (COMPLETED_FLAG != threadFlag.get())  {
```
Please revert this formatting change (opening braces inside functions are on same line and new line only for functions), it is as per Mbed OS coding guidelines.

Reply:
@nuertey use astyle and this Mbed OS AStyle configuration https://github.com/ARMmbed/mbed-os/blob/master/.astylerc